assets/src/stories-editor/blocks/amp-story-text/edit.js

```diff
@@ -146,30 +212,77 @@ class TextBlockEdit extends Component {
 				</BlockControls>
 				<div className={ classnames( wrapperClass, {
 					'with-line-height': ampFitText,
+					'is-empty-draggable-text': ! isEditing && ! content.length,
 				} ) } style={ wrapperStyle } >
-					<RichText
-						wrapperClassName="wp-block-amp-story-text"
-						tagName="p"
-						// Ensure line breaks are normalised to HTML.
-						value={ content }
-						onChange={ ( nextContent ) => setAttributes( { content: nextContent } ) }
-						// The 2 following lines are necessary for pasting to work.
-						onReplace={ this.onReplace }
-						onSplit={ () => {} }
-						style={ {
-							color: textColor.color,
-							fontSize: ampFitText ? autoFontSize + 'px' : userFontSize,
-							textAlign: align,
-							position: ampFitText && content.length ? 'static' : undefined,
-						} }
-						className={ classnames( className, {
-							'has-text-color': textColor.color,
-							[ textColor.class ]: textColor.class,
-							[ fontSize.class ]: ampFitText ? undefined : fontSize.class,
-							'is-amp-fit-text': ampFitText,
-						} ) }
-						placeholder={ placeholder || __( 'Write text…', 'amp' ) }
-					/>
+					{ isEditing &&
+						<RichText
+							wrapperClassName={ textWrapperClassName }
+							tagName="p"
+							// Ensure line breaks are normalised to HTML.
+							value={ content }
+							onChange={ ( nextContent ) => setAttributes( { content: nextContent } ) }
+							// The 2 following lines are necessary for pasting to work.
+							onReplace={ this.onReplace }
+							onSplit={ () => {} }
+							style={ textStyle }
+							className={ classnames( className, textClassNames ) }
+							placeholder={ placeholder || __( 'Write text…', 'amp' ) }
+						/>
+					}
+					{ ! isEditing &&
+						<StoryBlockMover
+							clientId={ clientId }
+							blockName={ name }
+							blockElementId={ `block-${ clientId }` }
+							isDraggable={ ! isPartOfMultiSelection }
+							isMovable={ true }
+						>
+							<div
+								role="textbox"
+								tabIndex="-1"
+								className="is-not-editing editor-rich-text block-editor-rich-text wp-block-amp-story-text"
+								onClick={ () => {
+									if ( isSelected ) {
+										this.toggleIsEditing( true );
+									}
+								} }
+								onKeyDown={ ( e ) => {
+									if ( ENTER === e.keyCode && isSelected ) {
+										this.toggleIsEditing( true );
+									}
```

Comment:
This isn't really working.

Reply:
Is the callback even called? Or is the condition not working? Or the toggling? Need `event.preventDefault();`?

Reply:
It's not even called.

I read that it might be for inputs mainly or when `contenteditable` is true, however, it does look like there are some places in Gutenberg where these are used without these conditions. Not sure.

We can't set `contenteditable` either since then the overlay would lose its purpose.

Any thoughts?
Added the eslint exception for now, the rule is `jsx-a11y/click-events-have-key-events`.

Reply:
I think you need a `tabindex` in order to be able to listen to `onKeyDown` on a `div`.

The `tabindex` could only be added when the block is selected.

Reply:
It actually does have a `tabindex` already. It had `tabindex` `0` before but since the key events didn't seem to be working anyway then set it to `-1`.

Also tried removing the `BlockMover` to see if that might be interfering with the events but still didn't work.

Also tested that `onKeyDown` within `block.js` is not triggered at all, it might be related to the issue here: https://github.com/ampproject/amp-wp/issues/2787#issuecomment-516988322

Thoughts?

Reply:
Just tested it again and discovered that in some cases it does work. Haven't understood yet in which cases it works and which not. 

I'll implement the keydown events for the working times 👍